shuup/gdpr/models.py

```diff
@@ -125,24 +140,47 @@ class Meta:
 
     @classmethod
     def create_for_user(cls, user, shop, consent_documents):
-        gdpr_user_consent = cls.objects.create(shop=shop, user=user)
         documents = []
-        for document in consent_documents:
-            version = Version.objects.get_for_object(document).first()
+        for page in consent_documents:
+            Page.create_initial_revision(page)
+            version = Version.objects.get_for_object(page).first()
             consent_document = GDPRUserConsentDocument.objects.create(
-                page=document,
+                page=page,
                 version=version
             )
             documents.append(consent_document)
+        gdpr_user_consent, created = cls.objects.get_or_create(shop=shop, user=user)
         gdpr_user_consent.documents = documents
         return gdpr_user_consent
 
-    def should_reconsent(self):
+    def should_reconsent(self, shop, user):
```

Comment:
could this something simpler? As you don't need to know the differences, you probably can achieve this by doing something like:

```py
consent_pages_ids = set([page.id for page in get_active_consent_pages(shop)])
docs_ids = set([page.id for page in self.documents.all()])
if consent_pages_ids != docs_ids:
   return True
```

makes sense? of course, you have to check version at the end too, maybe you can even create a set of tuples of `(document_id, version_id`) and then use that when comparing sets